Source/DafnyCore/DafnyFile.cs

```diff
@@ -17,23 +19,26 @@ public class DafnyFile {
   public string Extension { get; private set; }
   public string CanonicalPath { get; }
   public string BaseName { get; private set; }
-  public bool IsPreverified { get; private set; }
-  public bool IsPrecompiled { get; private set; }
+  public bool ShouldNotVerify { get; private set; }
+  public bool ShouldNotCompile { get; private set; }
   public DafnyOptions FileOptions { get; private set; }
   public Func<TextReader> GetContent { get; set; }
   public Uri Uri { get; private set; }
   public IToken? Origin { get; }
 
   private static readonly Dictionary<Uri, Uri> ExternallyVisibleEmbeddedFiles = new();
 
+  static DafnyFile() {
+    DooFile.RegisterLibraryCheck(UnsafeDependencies, OptionCompatibility.OptionLibraryImpliesLocalError);
```

Comment:
I assume the idea is to make a library built with this flag unusable?

This approach is effective at least, but as a future UX improvement we should probably error on the attempt to do `dafny build -t:lib --unsafe-dependencies` in the first place as well.

Reply:
Fair.